src/mapblock_mesh.cpp

```diff
@@ -335,12 +335,17 @@ u16 getSmoothLight(v3s16 p, v3s16 corner, MeshMakeData *data)
 void finalColorBlend(video::SColor& result,
 		u8 day, u8 night, u32 daynight_ratio)
 {
-	s32 rg = (day * daynight_ratio + night * (1000-daynight_ratio)) / 1000;
-	s32 b = rg;
+	s32 r = (day * daynight_ratio + night * (1000-daynight_ratio)) / 1000;
+	s32 g = r;
+	s32 b = r;
 
 	// Moonlight is blue
 	b += (day - night) / 13;
-	rg -= (day - night) / 23;
+	r -= (day - night) / 23;
+
+	// Morning/Afternoon is red
+	g -= MYMAX(0, 100 - abs(daynight_ratio - 500)) / 4;
+	b -= MYMAX(0, 100 - abs(daynight_ratio - 500)) / 4;
```

Comment:
`MYMAX(0, ...)` is equivalent to `abs(...)`, except that `MYMAX` evaluates its parameters twice.  Just use `abs`.